### ISSUE TITLE
request: import once functionality

### ISSUE THREAD

**Issue by TxHawks** [davidkpiano/sassport#10]
An import-once functionality in the default, reference and dynamic loaders would be really nice.

See [node-sass-import-once](https://github.com/at-import/node-sass-import-once) and [eyeglass's import-once](https://github.com/sass-eyeglass/eyeglass#import-once)

**Comment by davidkpiano** [comment]
I'll be releasing this soon! Here's a proof of concept: https://github.com/davidkpiano/sassport/blob/master/src/modules/once.js

You can actually straight up use this module right now:

``` js
var importOnce = require('sassport/lib/modules/once');

//...

sassport([ importOnce ]).render(...);
```

Then in your SCSS:

``` scss
@import 'some-partial !once';

// ...
```

**Comment by TxHawks** [comment]
Is it possible to define `once` as the default behavior? And is it possible to mix loaders?

I mean do something like

``` scss
@import 'some-partial !once !reference';
```

Sorry for the barrage of questions

**Comment by davidkpiano** [comment]
It's definitely possible to use multiple loaders, and they will run left to right: https://github.com/davidkpiano/sassport/blob/master/test/loaders.js#L52

So your example **will work**, and it will first run it through `once` and then through `reference`.

I can definitely see having a `defaultLoaders` configuration, what do you think?

``` js
sassport([ once, reference ], {
  defaultLoaders: ['once']
}).render(...);
```

**Comment by TxHawks** [comment]
Looks really good.

**Comment by TxHawks** [comment]
I assume the import-once functionality is not yet available in the version that's NPM, right?

**Comment by davidkpiano** [comment]
It actually is available, just not documented: https://github.com/davidkpiano/sassport/blob/master/test/loaders.js#L86

But I will be making a separate package for this module, most likely called `sassport-once`.

**Comment by TxHawks** [comment]
Would this work with sub-dependencies? I think this would be the most interesting use case, as it would enable efficient modules.

The scenario is in mind is this:

``` scss
// _module-a.scss
@import `node_modules/module-b`;

// styles.scss
@import `node_modules/module-a`;
@import `node_modules/module-b`;
```

ideally, an `import-once` loader should recognize the `module-b` was already imported inside `module-a` and not import it again in `style.scss`

**Comment by davidkpiano** [comment]
Yes, it will. In Sass, importing is a blocking synchronous action, meaning that this sequence of imports will happen:
- import `module-a` (current stack: `[module-a]`)
  - import `module-b` (current stack: `[module-a, module-b]`)
- skip `module-b` since it was already imported by checking current stack

**Comment by TxHawks** [comment]
:+1:
